TryAtSoftware.CleanTests.Core/Utilities/CombinatorialMachine.cs

```diff
@@ -37,60 +39,69 @@ void Dfs(int categoryIndex)
 
             var currentCategory = categories[categoryIndex];
 
-            var requiredCharacteristics = activeDemands.EnsureValue(currentCategory).Keys.ToArray();
             foreach (var utility in this._utilities.Get(currentCategory))
             {
-                var canBeUsed = utility.FulfillsAllDemands(requiredCharacteristics);
-                if (!canBeUsed || !RegisterExternalDemands(utility, slots, activeDemands)) continue;
+                if (incompatibilityFactorByUtilityId[utility.Id] > 0) continue;
 
+                foreach (var iu in incompatibleUtilitiesMap[utility.Id]) incompatibilityFactorByUtilityId[iu]++;
```

Comment:
This map may contain utilities that are irrelevant (from categories that are already included as a part of the current combination). This can be optimized a little more.

Reply:
Well, it looks like this is not worth of optimizing.